### PR TITLE
Update to use prop-types instead of React.PropTypes

### PR DESCRIPTION
**Pull request by binoy14**
In preparation for React 16, using to use `prop-types` package to use PropTypes

https://facebook.github.io/react/blog/#new-deprecation-warnings

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,18 +54,18 @@
     "eslint-plugin-react": "^6.10.0",
     "eslint-plugin-react-native": "^2.3.1",
     "jest": "^19.0.2",
+    "opencollective-postinstall": "^1.0.15",
```
Comment (Monte9) [inline package.json]: this already exists on master.

**Comment by Monte9** [comment]
💯 for keeping up with the latest changes with React.

Let's shoot to get this merged in within the next couple releases. Wanna make sure we are stable with propTypes. `v0.11.3-rc` possibly.

File: src/input/Search.js [hunk 2]
```diff
@@ -106,8 +107,8 @@ Search.propTypes = {
   showLoadingIcon: PropTypes.bool,
   loadingIcon: PropTypes.object,
   clearIcon: PropTypes.oneOf(PropTypes.object, PropTypes.bool),
-  textInputRef: PropTypes.string,
-  containerRef: PropTypes.string,
+  textInputRef: PropTypes.func,
```
Comment (fkrauthan) [inline src/input/Search.js]: You might wanna consider using `PropTypes. oneOfType([PropTypes.func, PropTypes.string])` to keep it backwards compatible. Valid for all other ref replacements.

**Comment by fkrauthan** [comment]
@Monte9 You are also missing ref propType validation changes for FormInput (and not sure if other files have that aswell. Worth doing a search in all of them)

**Comment by binoy14** [comment]
@tarang9211 I am closing this PR and making a new one since things have changed and it's worth fixing conflicts if it can be easily done a codemod. For future reference do not commit two different issues or features part of one PR. You can cherry-pick your commit to a different branch and make the PR. Also note the change @fkrauthan suggested.

**Pull request closed without merging**